core/pom.xml

```diff
@@ -243,6 +243,12 @@
             <artifactId>bson</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.example</groupId>
+            <artifactId>solver</artifactId>
+            <version>3.0.1-SNAPSHOT</version>
```

Comment:
this should not be here.
it should be in root pom file under `dependencyManagement`, using `${project.version}`. 
here, in submodule, should be included without version (as it will be inherited)